src/CoreWCF.Primitives/src/CoreWCF/ServiceBehaviorAttribute.cs

```diff
@@ -249,6 +274,7 @@ private void ApplyInstancing(ServiceDescription description, ServiceHostBase ser
                                 }
 
                                 singleton.AutoClose = false;
+                                singleton.Extensions.Add(new ServiceProviderExtension(_serviceProvider));
                             }
```

Comment:
Would it be worth adding a null check here? If this is expected to always not be null (which as far as I can tell is the case), can you add an Fx.Assert so we fail during testing if this isn't the case.

Reply:
> Just needs a validation assert on the service provider then this is ready to merge. This is great that we're now expanding CoreWCF to go beyond what WCF could do.

@mconnew  I just pushed #493 There'is is still a lot to do in the generator but that's a starting point

Reply:
done